plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -127,7 +127,20 @@ func (g *genericScheduler) Schedule(pod *v1.Pod, nodeLister algorithm.NodeLister
 	}
 
 	trace.Step("Selecting host")
-	return g.selectHost(priorityList)
+	host, err := g.selectHost(priorityList)
+	if err != nil {
+		return "", err
+	}
+
+	if len(g.extenders) != 0 {
+		for _, extender := range g.extenders {
+			err := extender.Bind(pod, host)
```

Comment:
It is the same recovery path. k8s scheduler will schedule the pod again which will result in another Bind for the same pod UID. That can be used by the extender to free up resources allocated earlier (or reuse them if the pod is being bound the same node).